### PR TITLE
Merge robot_state_publisher back to original repository.

### PR DESCRIPTION
**Pull request by clalancette**
Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

### CODE REVIEW

File: ros2.repos [hunk 1]
```diff
@@ -257,7 +257,7 @@ repositories:
     version: master
   ros2/robot_state_publisher:
```
Comment (dirk-thomas) [inline ros2.repos]: Should the target directory be updated too?

Comment (clalancette) [reply ros2.repos]: Yeah, it wasn't totally clear to me if we should move it back to the `ros` directory or not.  I don't have a strong opinion either way.

Comment (dirk-thomas) [reply ros2.repos]: If there is no strong reason not to I would suggest to move it to the matching directory location.

Comment (Karsten1987) [reply ros2.repos]: addressed in here: https://github.com/ros2/ros2/pull/803

**Pull request merged**